packages/flutter/lib/src/material/typography.dart

```diff
@@ -47,58 +45,98 @@ class TextTheme extends Diagnosticable {
   /// If you do decide to create your own text theme, consider using one of
   /// those predefined themes as a starting point for [copyWith] or [apply].
   const TextTheme({
-    this.display4,
-    this.display3,
-    this.display2,
-    this.display1,
-    this.headline,
-    this.title,
-    this.subhead,
-    this.body2,
-    this.body1,
+    TextStyle display4,
+    TextStyle display3,
+    TextStyle display2,
+    TextStyle display1,
+    TextStyle headline,
+    TextStyle title,
+    TextStyle subhead,
+    TextStyle body2,
+    TextStyle body1,
     this.caption,
     this.button,
-  });
+    TextStyle headline1,
+    TextStyle headline2,
+    TextStyle headline3,
+    TextStyle headline4,
+    TextStyle headline5,
+    TextStyle headline6,
+    TextStyle subtitle1,
+    this.subtitle2,
+    TextStyle paragraph1,
+    TextStyle paragraph2,
+    this.overline,
+  }) : assert(!(display4 != null && headline1 != null)),
+       assert(!(display3 != null && headline2 != null)),
+       assert(!(display2 != null && headline3 != null)),
+       assert(!(display1 != null && headline4 != null)),
+       assert(!(headline != null && headline5 != null)),
+       assert(!(title != null && headline6 != null)),
+       assert(!(subhead != null && subtitle1 != null)),
+       assert(!(body1 != null && paragraph2 != null)),
+       assert(!(body2 != null && paragraph1 != null)),
+       this.headline1 = headline1 ?? display4,
+       this.headline2 = headline2 ?? display3,
+       this.headline3 = headline3 ?? display2,
+       this.headline4 = headline4 ?? display1,
+       this.headline5 = headline5 ?? headline,
+       this.headline6 = headline6 ?? title,
+       this.subtitle1 = subtitle1 ?? subhead,
+       this.paragraph2 = paragraph2 ?? body1,
+       this.paragraph1 = paragraph1 ?? body2;
 
   /// Extremely large text.
   ///
   /// The font size is 112 pixels.
-  final TextStyle display4;
+  TextStyle get display4 => headline1;
```

Comment:
i think we should go even further and just not have "headline1" in our API. We should IMHO just have a table in the class API docs that has the mapping from the names in the spec to the names in our API.

Reply:
(mostly i'm worried that this won't scale. Two years from now, if we have a third set of names, do we add those too? What if they overlap?)

Reply:
I don't think we'll ever be able to get rid of the old names although we can eliminate their use in the Flutter implementation. It seems reasonable, albeit painful, to sync up with the new names once. But never again.